lib/posix/pthread.c

```diff
@@ -14,16 +14,15 @@
 #include <zephyr/sys/slist.h>
 
 #include "posix_internal.h"
```

Comment:
Just a note, but the `posix_internal.h` header is unfortunately sensitive to include order 😢 - this should be fixed at some point (#57232)